### DISCUSSION TITLE
Windows vs WindowsApps wsl.exe

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/WSL#10067] [General] [answered]
Goal: understand WSL execution environment/arch.

What is the difference between the following?
- C:\WINDOWS\system32\wsl.exe
- C:\Program Files\WindowsApps\MicrosoftCorporationII.WindowsSubsystemForLinux_1.2.5.0_x64__8wekyb3d8bbwe\wsl.exe

Is one of them invalid/deprecated and should I delete one?

**Comment by Biswa96** [comment]
> Is one of them invalid/deprecated?

The system32 one will be removed after some years probably. Or it will become just a wrapper over the WindowsApps one.

> should I delete one?

Never modify any system file of Microsoft Windows operating system.

**Comment by benhillis** [comment, marked as answer]
No they are both still valid. When the WSL app is installed, the one in system32 will redirect to the packaged version.
